prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -524,6 +541,26 @@ private void setEntityTags(Entity entity, String tags) {
         }
     }
 
+    private  JSONObject buildDetailResult(final String extensionName) throws FalconException {
+        ExtensionMetaStore metaStore = ExtensionStore.get().getMetaStore();
+
+        if (!metaStore.checkIfExtensionExists(extensionName)){
+            throw new StoreAccessException(new Exception("No extension resources found for " + extensionName));
```

Comment:
Unnecessary nested exception creation. You can throw ValidationException instead.